src/wallet/wallet.cpp

```diff
@@ -3626,9 +3685,12 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
             CPubKey masterPubKey = walletInstance->GenerateNewHDMasterKey();
             if (!walletInstance->SetHDMasterKey(masterPubKey))
                 throw std::runtime_error(std::string(__func__) + ": Storing master key failed");
+
+            // ensure this wallet.dat can only be opened by clients supporting HD with chain split
```

Comment:
Technically this should happen before SetHDMasterKey, I believe (otherwise it maybe a race?)